python3Packages.llfuse fails to build on Darwin

**Describe the bug**
borgbackup fails to build on unstable.
I've included the stacktrace.
It looks related to llfuse.
I'm on darwin but I'm not sure it's related.
I'm currently trying to build llfuse on nixpkgs master to verify the problem.

**To Reproduce**
Steps to reproduce the behavior:
1. ...
2. ...
3. ...

**Additional context**
stacktrace
```
warning: Git tree '/Users/raphael/.dotfiles' is dirty
error: --- Error --- nix-daemon
builder for '/nix/store/naxs86vlf9jjc703q29n8adzfv4rwhz8-python3.8-llfuse-1.3.8.drv' failed with exit code 2; last 10 log lines:
  test/util.py:36
    /private/tmp/nix-build-python3.8-llfuse-1.3.8.drv-0/llfuse-1.3.8/test/util.py:36: PytestUnknownMarkWarning: Unknown pytest.mark.uses_fuse - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
      return pytest.mark.uses_fuse()

  -- Docs: https://docs.pytest.org/en/stable/warnings.html
  =========================== short test summary info ============================
  ERROR test/test_api.py
  ERROR test/test_fs.py
  !!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
  ================== 3 deselected, 1 warning, 2 errors in 2.29s ==================
error: --- Error --- nix-daemon
1 dependencies of derivation '/nix/store/lqlaf9yczs3mx1bjlh8r15ccrcdkzkj6-borgbackup-1.1.16.drv' failed to build
error: --- Error --- nix-daemon
1 dependencies of derivation '/nix/store/1aw5nhd0kmz5qyb2nh1y9mv1wlgy23d0-system-applications.drv' failed to build
error: --- Error ------------------------------------------------------------------------------------- nix
1 dependencies of derivation '/nix/store/ghdm3zi57s56b25mqiac1p2fk8mski21-darwin-system-21.05.20210325.c0e8818+darwin4.1a76093.drv' failed to build
```

**Notify maintainers**
@flokli @dotlambda @globin 
@bjornfor 

Sorry if I'm pinging too many people.

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

Here is the full stack trace building llfuse
```
Executing pipInstallPhase
/private/tmp/nix-build-python2.7-llfuse-1.3.8.drv-0/llfuse-1.3.8/dist /private/tmp/nix-build-python2.7-llfuse-1.3.8.drv-0/llfuse-1.3.8
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
Processing ./llfuse-1.3.8-cp27-cp27mu-macosx_10_12_x86_64.whl
Requirement already satisfied: contextlib2 in /nix/store/jlwfi6wljn54mi7q0i00aqfji67s13wg-python2.7-contextlib2-0.6.0.post1/lib/python2.7/site-packages (from llfuse==1.3.8) (0.6.0.post1)
Installing collected packages: llfuse
Successfully installed llfuse-1.3.8
/private/tmp/nix-build-python2.7-llfuse-1.3.8.drv-0/llfuse-1.3.8
Finished executing pipInstallPhase
post-installation fixup
strip is /nix/store/z8d39gw4q90i97jlsilbad33siwh4yxd-cctools-binutils-darwin-949.0.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/w4q7311alq7szczh2jhrd5vzwv7ldsbq-python2.7-llfuse-1.3.8/lib
patching script interpreter paths in /nix/store/w4q7311alq7szczh2jhrd5vzwv7ldsbq-python2.7-llfuse-1.3.8
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
============================= test session starts ==============================
platform darwin -- Python 2.7.18, pytest-4.6.11, py-1.9.0, pluggy-0.13.1
rootdir: /private/tmp/nix-build-python2.7-llfuse-1.3.8.drv-0/llfuse-1.3.8
collected 3 items / 2 errors / 3 deselected

==================================== ERRORS ====================================
______________________ ERROR collecting test/test_api.py _______________________
ImportError while importing test module '/private/tmp/nix-build-python2.7-llfuse-1.3.8.drv-0/llfuse-1.3.8/test/test_api.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
test/test_api.py:19: in <module>
    import llfuse
E   ImportError: dlopen(/nix/store/w4q7311alq7szczh2jhrd5vzwv7ldsbq-python2.7-llfuse-1.3.8/lib/python2.7/site-packages/llfuse.so, 2): Library not loaded: /usr/local/lib/libfuse.2.dylib
E     Referenced from: /nix/store/w4q7311alq7szczh2jhrd5vzwv7ldsbq-python2.7-llfuse-1.3.8/lib/python2.7/site-packages/llfuse.so
E     Reason: image not found
_______________________ ERROR collecting test/test_fs.py _______________________
ImportError while importing test module '/private/tmp/nix-build-python2.7-llfuse-1.3.8.drv-0/llfuse-1.3.8/test/test_fs.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
test/test_fs.py:19: in <module>
    import llfuse
E   ImportError: dlopen(/nix/store/w4q7311alq7szczh2jhrd5vzwv7ldsbq-python2.7-llfuse-1.3.8/lib/python2.7/site-packages/llfuse.so, 2): Library not loaded: /usr/local/lib/libfuse.2.dylib
E     Referenced from: /nix/store/w4q7311alq7szczh2jhrd5vzwv7ldsbq-python2.7-llfuse-1.3.8/lib/python2.7/site-packages/llfuse.so
E     Reason: image not found
=============================== warnings summary ===============================
/nix/store/3rr3442ggj25ssf5y969ma96r59k75kq-python2.7-pytest-4.6.11/lib/python2.7/site-packages/_pytest/mark/structures.py:335
  /nix/store/3rr3442ggj25ssf5y969ma96r59k75kq-python2.7-pytest-4.6.11/lib/python2.7/site-packages/_pytest/mark/structures.py:335: PytestUnknownMarkWarning: Unknown pytest.mark.uses_fuse - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

-- Docs: https://docs.pytest.org/en/latest/warnings.html
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
============== 3 deselected, 1 warnings, 2 error in 0.76 seconds ===============
error: --- Error ------------------------------------------------------------------------------- nix-build
builder for '/nix/store/wnyisar7i7jfbycrk4gkxy8hdw7i8lm7-python2.7-llfuse-1.3.8.drv' failed with exit code 2; last 10 log lines:
  E     Referenced from: /nix/store/w4q7311alq7szczh2jhrd5vzwv7ldsbq-python2.7-llfuse-1.3.8/lib/python2.7/site-packages/llfuse.so
  E     Reason: image not found
  =============================== warnings summary ===============================
  /nix/store/3rr3442ggj25ssf5y969ma96r59k75kq-python2.7-pytest-4.6.11/lib/python2.7/site-packages/_pytest/mark/structures.py:335
    /nix/store/3rr3442ggj25ssf5y969ma96r59k75kq-python2.7-pytest-4.6.11/lib/python2.7/site-packages/_pytest/mark/structures.py:335: PytestUnknownMarkWarning: Unknown pytest.mark.uses_fuse - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
      PytestUnknownMarkWarning,

  -- Docs: https://docs.pytest.org/en/latest/warnings.html
  !!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
  ============== 3 deselected, 1 warnings, 2 error in 0.76 seconds ===============
```

--- a/pkgs/development/python-modules/llfuse/default.nix
+++ b/pkgs/development/python-modules/llfuse/default.nix
@@ -35,12 +35,12 @@ buildPythonPackage rec {
     ${python.interpreter} setup.py build_cython
   '';
 
+  # On Darwin, the test requires macFUSE to be installed outside of Nix.
+  doCheck = !stdenv.isDarwin;
   checkInputs = [ pytestCheckHook which ];
 
   disabledTests = [
     "test_listdir" # accesses /usr/bin
-  ] ++ lib.optionals stdenv.isDarwin [
-    "uses_fuse"
   ];
 
   meta = with lib; {

